### PR TITLE
wantlist command

### PR DESCRIPTION
**Pull request by whyrusleeping**
I wanted to be able to check my wantlist when having trouble resolving things to get a better idea of what was going on. So I made this. not sure what tests to write for it, so if you have any ideas, let me know.

### CODE REVIEW

**Comment by jbenet** [comment]
This is a bitswap specific thing. lets put it under  `ipfs bitswap wantlist`

File: core/commands/bitswap.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	},
+	Type: KeyList{},
+	Run: func(req cmds.Request, res cmds.Response) {
+		nd, err := req.Context().GetNode()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		res.SetOutput(&KeyList{nd.Exchange.GetWantlist()})
```
Comment (jbenet) [inline core/commands/bitswap.go]: this should still cast the exchange to bitswap, as this is a command called bitswap :)

File: exchange/interface.go [hunk 2]
```diff
@@ -22,5 +22,7 @@ type Interface interface {
 	// available on the network?
 	HasBlock(context.Context, *blocks.Block) error
 
+	GetWantlist() []u.Key
```
Comment (jbenet) [inline exchange/interface.go]: let's not add this here. not all exchanges will have a wantlist (think of HTTP as a possible exchange)

File: exchange/offline/offline.go [hunk 3]
```diff
@@ -66,3 +66,8 @@ func (e *offlineExchange) GetBlocks(ctx context.Context, ks []u.Key) (<-chan *bl
 	}()
 	return out, nil
 }
+
+// implement Exchange
+func (e *offlineExchange) GetWantlist() []u.Key {
+	return nil
+}
```
Comment (jbenet) [inline exchange/offline/offline.go]: unneeded given https://github.com/jbenet/go-ipfs/pull/719/files#r25191683

**Comment by whyrusleeping** [comment]
@jbenet  is this RFM?

**Comment by jbenet** [comment]
@whyrusleeping can you rebase and push again? looks like travis is still complaining about init docs hash (that got fixed, right?) 

and, looks like there's a race?
- https://build.protocol-dev.com/job/race/17293/console

This LGTM, except tests dont seem to be passing

**Comment by whyrusleeping** [comment]
The race doesnt appear to be related to the changes in this PR, I filed a bug for it.

**Comment by jbenet** [comment]
@whyrusleeping what about https://build.protocol-dev.com/job/epic/1257/console ? it ran on cassie-- is that a timeout or oom?

**Comment by whyrusleeping** [comment]
That appears to be a timeout, looks like the "bitswap no routing" tests can hang...

**Comment by jbenet** [comment]
This is new though. They weren't hanging before. Could it be the bitswap changes last week?




Understand we want to hurry on this PR and ipns but if we let bugs in it will be really hard to squash them later.



—

**Comment by whyrusleeping** [comment]
Yeah, ive seen it twice before. And I think it has happened since my bitswap changes. Ill look into it.

**Comment by whyrusleeping** [comment]
This is probably caused by my latest bitswap changes, and I think I maybe understand the issue. Ill delve into it more when i get a little more time.

**Pull request merged**